pkg/server/server_authz_test.go

```diff
@@ -0,0 +1,1570 @@
+			define can_call_expand: [application] or reader
+			define can_call_list_objects: [application] or reader
+			define can_call_list_users: [application] or reader
+			define can_call_read: [application] or reader
+			define can_call_read_assertions: [application] or reader or model_writer
+			define can_call_read_authorization_models: [application] or reader or model_writer
+			define can_call_read_changes: [application] or reader
+			define can_call_write: [application] or writer
+			define can_call_write_assertions: [application] or model_writer
+			define can_call_write_authorization_models: [application] or model_writer
+			define model_writer: [application] or admin
+			define reader: [application] or admin
+			define writer: [application] or admin
+			define admin: [application] or creator or admin from system
+		`
+	testStoreModel = `
+			model
+				schema 1.1
+
+			type channel
+			relations
+				define commenter: [user, workspace#member] or writer
+				define parent_workspace: [workspace]
+				define writer: [user, workspace#member]
+
+			type user
+
+			type workspace
+			relations
+				define channels_admin: [user] or legacy_admin
+				define guest: [user]
+				define legacy_admin: [user]
+				define member: [user] or legacy_admin or channels_admin
+		`
+)
+
+func newSetupAuthzModelAndTuples(t *testing.T, openfga *Server, clientID string) *authzSettings {
+	rootStore, err := openfga.CreateStore(context.Background(), &openfgav1.CreateStoreRequest{Name: "root-store"})
+	require.NoError(t, err)
+
+	writeAuthzModelResp, err := openfga.WriteAuthorizationModel(context.Background(), &openfgav1.WriteAuthorizationModelRequest{
+		StoreId:         rootStore.GetId(),
+		TypeDefinitions: language.MustTransformDSLToProto(rootStoreModel).GetTypeDefinitions(),
+		SchemaVersion:   typesystem.SchemaVersion1_1,
+	})
+	require.NoError(t, err)
+
+	rootStoreModelID := writeAuthzModelResp.GetAuthorizationModelId()
+
+	_, err = openfga.Write(context.Background(), &openfgav1.WriteRequest{
+		StoreId:              rootStore.GetId(),
+		AuthorizationModelId: rootStoreModelID,
+		Writes: &openfgav1.WriteRequestWrites{
+			TupleKeys: []*openfgav1.TupleKey{
+				tuple.NewTupleKey(fmt.Sprintf("store:%s", rootStore.GetId()), "admin", fmt.Sprintf("application:%s", clientID)),
+			},
+		},
+	})
+	require.NoError(t, err)
+
+	testStore, err := openfga.CreateStore(context.Background(), &openfgav1.CreateStoreRequest{Name: "test-store"})
+	require.NoError(t, err)
+
+	writeTestStoreAuthzModelResp, err := openfga.WriteAuthorizationModel(context.Background(), &openfgav1.WriteAuthorizationModelRequest{
+		StoreId:         testStore.GetId(),
+		TypeDefinitions: language.MustTransformDSLToProto(testStoreModel).GetTypeDefinitions(),
+		SchemaVersion:   typesystem.SchemaVersion1_1,
+	})
+	require.NoError(t, err)
+
+	testStoreModelID := writeTestStoreAuthzModelResp.GetAuthorizationModelId()
+
+	return &authzSettings{
+		openfga:  openfga,
```

Comment:
Nitpick: why do you store a reference to the server? The caller gave you the reference, it doesn't need it back, right?

Reply:
It's so that `addAuthForRelation` can easily access the server. Maybe there is a better way to do it?

Reply:
@miparnisari - do you have a suggestion for this or is this something we can merge now and fix in the final PR?

Reply:
Yep definitely. I don't have a suggestion atm